examples/vision/sample_size_estimate.py

```diff
@@ -0,0 +1,545 @@
+
+    Returns:
+        Uncompiled Keras model.
+    """
+
+    # Create input and pre-processing layers for MobileNetV2
+    inputs = layers.Input(shape=(img_size, img_size, 3))
+    x = preprocessing.Rescaling(scale=1.0 / 127.5, offset=-1)(inputs)
+    model = keras.applications.MobileNetV2(
+        include_top=False, weights="imagenet", input_tensor=x
+    )
+
+    # Freeze the pretrained weights
+    model.trainable = False
+
+    # Rebuild top
+    x = layers.GlobalAveragePooling2D(name="avg_pool")(model.output)
+    x = layers.Dropout(top_dropout)(x)
+    outputs = layers.Dense(num_classes, activation="softmax")(x)
+    model = tf.keras.Model(inputs, outputs)
+
+    print("Trainable weights:", len(model.trainable_weights))
+    print("Non_trainable weights:", len(model.non_trainable_weights))
+    return model
+
+
+def compile_and_train(
+    model,
+    train_ds,
+    val_ds,
+    metrics=[keras.metrics.AUC(name="auc"), "acc"],
+    optimizer=keras.optimizers.Adam(),
+    patience=5,
+    epochs=5,
+):
+    """Compiles and trains a Keras model using EarlyStopping callback on
+        'val_auc' (requires at minimum 'auc' as a metric. Compiles using
+        categorical_crossentropy loss with optimizer and metrics of choice.
+
+    Arguments:
+        model: Uncompiled Keras model.
+        train_ds: tf.data.Dataset, trainig dataset.
+        val_ds: tf.data.Dataset, validation dataset.
+        class_weights: Dict, weights per class.
+        metrics: Keras/TF metrics, requires at least 'auc' metric(defaults is
+                [keras.metrics.AUC(name='auc'), 'acc']).
+        optimizer: Keras/TF optimizer (defaults is keras.optimizers.Adam()).
+        patience: Int, epochsfor EarlyStopping patience (defaults is 5).
+        epochs: Int, number of epochs to train (default is 5).
+
+    Returns:
+        Training history for trained Keras model.
+    """
+
+    stopper = keras.callbacks.EarlyStopping(
+        monitor="val_auc",
+        mode="max",
+        min_delta=0,
+        patience=patience,
+        verbose=1,
+        restore_best_weights=True,
+    )
+
+    model.compile(loss="categorical_crossentropy", optimizer=optimizer, metrics=metrics)
+
+    history = model.fit(
+        train_ds, epochs=epochs, validation_data=val_ds, callbacks=[stopper]
+    )
+    return history
+
+
+def unfreeze(model, block_name, verbose=0):
+    """Unfreezes Keras model layers.
+
+    Arguments:
+        model: Keras model.
+        block_name: Str, layer name for example block_name = 'block4'.
+                    Checks if supplied string is in the layer name.
+        verbose: Int, 0 means silent, 1 prints out layers trainability status.
+
+    Returns:
+        Keras model with all layers after (and including) the specified
+        block_name to trainable, excluding BatchNormalization layers.
+    """
+
+    # Set the whole model trainable and
+    model.trainable = True
+    set_trainable = False
```

Comment:
It appears you don't need this variable

Reply:
It is required as the function sets it to True when it reaches the designated layer. It should keep setting all subsequent layers to trainable even if the layer name does not contain the block_name.

Reply:
I do however, notice that I don't need to set model.trainable = True first, I can drop this and avoid setting layers to "not trainable" again.